--- a/kernel/nvidia/os-mlock.c
+++ b/kernel/nvidia/os-mlock.c
@@ -39,7 +39,7 @@
 #if defined(NV_FOLLOW_PFN_PRESENT)
 static int __nv_follow_pfn(struct vm_area_struct *vma, unsigned long address,
                             unsigned long *pfn) {
-    return follow_pfn(vma, address, pfn);
+    return follow_pte(vma, address, pfn);
 }
 #else
 static int __nv_follow_pfn(struct vm_area_struct *vma, unsigned long address,
